### PR TITLE
Update cadl to TypeSpec

### PR DESCRIPTION
**Pull request by chidozieononiwu**
CADL to TypeSpec rename. Should resolve https://github.com/Azure/azure-sdk-tools/issues/5635

Blocked by:
Updates to https://github.com/Azure/azure-sdk-tools/blob/7c10ce5ef4641263b81e775aa181d82750a83da1/src/dotnet/APIView/APIViewWeb/Languages/CadlLanguageService.cs
Updated to DataBase

### CODE REVIEW

**Review by tjprescott — CHANGES_REQUESTED** [review]
I see all the updates from Cadl to TypeSpec, but if someone selects "TypeSpec" the filters should still include any existing APIViews that have a language of type "Cadl"

**Comment by chidozieononiwu** [comment]
> I see all the updates from Cadl to TypeSpec, but if someone selects "TypeSpec" the filters should still include any existing APIViews that have a language of type "Cadl"

For that I plan to run a script to update all cadl values in the database. That will happen before this change is deployed.

**Comment by chidozieononiwu** [comment]
Do we want to have only `TypeSpec` or `TypeSpec | Cadl` on the UI side to accomodate teams that are not yet converting to TypeSpec @weshaggard @tjprescott

**Comment by tjprescott** [comment]
My strong recommendation would be to support `Cadl` as an alias of `TypeSpec`. For example, only choose `TypeSpec` in the filter, but it should show APIViews that are marked as langauge `TypeSpec` or `Cadl`. While a one-time conversion will "fix" any Cadl views currently in the system, if another gets uploaded it will just be silently swallowed by the void.

**Comment by weshaggard** [comment]
If using an alias (i.e. when querying the DB you can query for both TypeSpec and Cadl) is easy enough I think that is reasonable. I wouldn't display Cadl in the UI any longer.

**Comment by chidozieononiwu** [comment]
> If using an alias (i.e. when querying the DB you can query for both TypeSpec and Cadl) is easy enough I think that is reasonable. I wouldn't display Cadl in the UI any longer.

I have gone with this approach.

File: src/dotnet/APIView/APIViewWeb/Helpers/LanguageServiceHelpers.cs [hunk 1]
```diff
@@ -1,7 +1,27 @@
+using System.Collections.Generic;
+using System.Linq;
+
 namespace APIViewWeb.Helpers
 {
     public class LanguageServiceHelpers
     {
-        public static string[] SupportedLanguages = new string[] { "C", "C++", "C#", "Cadl", "Go", "Java", "JavaScript", "Json", "Kotlin", "Python", "Swagger", "Swift", "Xml" };
+        public static string[] SupportedLanguages = new string[] { "C", "C++", "C#", "TypeSpec", "Go", "Java", "JavaScript", "Json", "Kotlin", "Python", "Swagger", "Swift", "Xml" };
```
Comment (maririos) [inline src/dotnet/APIView/APIViewWeb/Helpers/LanguageServiceHelpers.cs:8]: nit: order of `TypeSpec`

**Review by maririos — APPROVED** [review]

**Pull request merged**